lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/launch/JavaLaunchCommandBuilder.java

```diff
@@ -184,11 +185,17 @@ public JavaLaunchCommandBuilder setJavaPath(final String path) {
 
   public JavaLaunchCommandBuilder setClassPath(final String classPath) {
     this.classPath = classPath;
+    if (SystemUtils.IS_OS_WINDOWS) {
```

Comment:
Hm... I think this in essence puts this class back to what it was before I touched it. I don't see anything wrong here, but if there is no change in behavior, maybe we should roll it back to this version.https://github.com/ryumbra/reef/blob/b7a98d795ca2fa1714da21640f3f1fc19d4c0a4d/lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/launch/JavaLaunchCommandBuilder.java

Reply:
The problem here is EvaluatorProcess is actually calling JavaLaunchCommandBuilder internally. By removing this, the class path from EvaluatorProcess.getCommandLine() will not have ' ' for Windows.